### PR TITLE
Implemented MFTF test extends reference and completion

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
### Description (*)
Implemented reference navigation and completion for the test extends XML attribute. Added test coverage for the functionality.

out

### Fixed Issues (if relevant)
1. magento/magento2-phpstorm-plugin#107: MFTF: Test name autocomplete and reference

### CODE REVIEW

File: resources/META-INF/plugin.xml [hunk 1]
```diff
@@ -105,7 +105,7 @@
         <fileBasedIndex implementation="com.magento.idea.magento2plugin.stubs.indexes.mftf.ActionGroupIndex" />
         <fileBasedIndex implementation="com.magento.idea.magento2plugin.stubs.indexes.mftf.DataIndex" />
         <fileBasedIndex implementation="com.magento.idea.magento2plugin.stubs.indexes.mftf.PageIndex" />
-        <fileBasedIndex implementation="com.magento.idea.magento2plugin.stubs.indexes.mftf.StepKeyIndex" />
```
Comment (VitaliyBoyko) [inline resources/META-INF/plugin.xml:108]: the Step key functionality is broken, will be re-implemented in future releases.

File: src/com/magento/idea/magento2plugin/linemarker/php/PluginTargetLineMarkerProvider.java [hunk 2]
```diff
@@ -169,6 +169,9 @@ private String getTargetMethodName(Method pluginMethod, String pluginPrefix) {
             String pluginMethodName = pluginMethod.getName();
             String targetClassMethodName = pluginMethodName.
                     replace(pluginPrefix, "");
+            if (targetClassMethodName.isEmpty()) {
```
Comment (VitaliyBoyko) [inline src/com/magento/idea/magento2plugin/linemarker/php/PluginTargetLineMarkerProvider.java:172]: I've faced the issue in the console and fixed it

File: src/com/magento/idea/magento2plugin/completion/provider/mftf/TestNameCompletionProvider.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+            ProcessingContext context,
+            @NotNull CompletionResultSet result) {
+        PsiElement position = parameters.getPosition().getOriginalElement();
+
+        if (position == null) {
+            return;
+        }
+
+        Collection<String> allKeys
+            = FileBasedIndex.getInstance().getAllKeys(TestNameIndex.KEY, position.getProject());
```
Comment (eduard13) [inline src/com/magento/idea/magento2plugin/completion/provider/mftf/TestNameCompletionProvider.java:34]: Lets keep this one in one line
```suggestion
        Collection<String> allKeys = FileBasedIndex.getInstance().getAllKeys(TestNameIndex.KEY, position.getProject());
```

**Comment by VitaliyBoyko** [comment]
Hi @eduard13 
Thank you!
I've excluded the current test name:
image

**Pull request merged**